### ISSUE TITLE
Automatically parallelise test suites

### ISSUE THREAD

**Issue by davidbanham** [Prismatik/redbeard#141]
If we Docker all the Dockers, we can take advantage of multicore machines by running parts of the test suite in parallel.

This can give us speedier test suites For Free! Mocks and things can still be used to speed up individual cases, but parallelisation is a rising tide that will lift all boats.

I've got a quick POC here:

https://github.com/Prismatik/shortlyster/commit/d05bc812eb08ec89e6d22416582c53376a655727

Does anyone see any icebergs in the path of this ship or should I continue the voyage?

**Comment by nwinch** [comment]
No icebergs that I can see.  Increasing the test suite speed is definitely a good thing!

Though, I am concerned that this ship could possibly sail past deeper issues.  And once sailed past, we might forget to come back.

Just wanted to flag this as it's in the same ocean as my concerns and I will post a clear discussion topic in a separate issue.

**Comment by MadRabbit** [comment]
as long as you run them in separate dbs it should be fine. although it's going to be tricky to merge back things like test coverage and failures tracking

**Comment by MadRabbit** [comment]
is it a problem though? afaik tests run under 30 seconds at the moment. i don't see why bother

**Comment by davidbanham** [comment]
It would definitely make reporting on failure less clear. You'd still get it in the output, but you'd probably need to scroll up in order to see it. If you were running down a single failure you'd probably have it in single mode and be using .only to speed up your TDD anyway.

The suite for redbeard is quick enough. This is in response to and end user finding that with about 7 models scaffolded out the suite started getting a bit slow.
